core/services/pg/event_broadcaster.go

```diff
@@ -262,7 +262,9 @@ func (sub *subscription) processQueue() {
 		select {
 		case sub.chEvents <- event:
 		case <-ctx.Done():
+			return
 		case <-sub.chDone:
+			return
```

Comment:
change ctx creation as following:
`ctx, cancel := utils.ContextFromChanWithDeadline(sub.chDone, broadcastTimeout)`
and remove the odd `case`.